lib/issuer.js

```diff
@@ -52,9 +64,107 @@ exports.issue = async ({instance, credential}) => {
   credential.issuanceDate = date.substr(0, date.length - 5) + 'Z';
   const verifiableCredential = await vc.issue({
     credential,
-    documentLoader: brJsonldDocumentLoader.documentLoader,
+    documentLoader,
     suite
   });
 
   return verifiableCredential;
 };
+
+exports.issueNew = async ({credential, profileAgentRecord}) => {
+  const {profileAgent} = profileAgentRecord;
+  const {zcaps: {
+    userDocument: userDocZcap,
+    userKak: userKakZcap,
+  }} = profileAgent;
+
+  const invocationSigner = await profileAgents.getSigner({profileAgentRecord});
+  const {httpsAgent} = brHttpsAgent;
+  const edvDocument = new EdvDocument({
+    capability: userDocZcap,
+    client: new EdvClient({httpsAgent}),
+    keyAgreementKey: kms.createKak({invocationSigner, zcap: userKakZcap}),
+    invocationSigner
+  });
+
+  const userDoc = await edvDocument.read();
+  const {content: {zcaps: {
+    ['key-assertionMethod']: assertionMethodZcap,
+    ['credential-edv-documents']: credentialsEdvZcap,
+    ['credential-edv-hmac']: credentialsEdvHmacZcap,
+    ['credential-edv-kak']: credentialsEdvKakZcap,
+  }}} = userDoc;
+
+  if(!(assertionMethodZcap && credentialsEdvZcap && credentialsEdvHmacZcap &&
+    credentialsEdvKakZcap)) {
+    throw new BedrockError('Permission denied.', 'NotAllowedError', {
+      httpStatusCode: 400,
+      public: true,
+    });
+  }
+  const {invocationTarget: {verificationMethod}} = assertionMethodZcap;
+  const issuerKey = kms.createAsymmetricKey(
+    {invocationSigner, zcap: assertionMethodZcap});
+
+  const suite = new Ed25519Signature2018({
+    verificationMethod,
+    signer: issuerKey
+  });
+
+  // FIXME: How do is this supposed to work? Is it just a matter of
+  // removing the hash fragment?
+  const issuer = verificationMethod;
+  credential.issuer = issuer;
+
+  const verifiableCredential = await vc.issue({
+    credential,
+    documentLoader,
+    suite
+  });
+
+  const edvClient = new EdvClient({
+    httpsAgent,
+    // FIXME: REQUIRED?
+    // id: edvId,
+    keyResolver: _keyResolver,
+    keyAgreementKey: new KeyAgreementKey({
+      id: credentialsEdvKakZcap.invocationTarget.id,
+      type: credentialsEdvKakZcap.invocationTarget.type,
+      capability: credentialsEdvKakZcap,
+      invocationSigner,
+      kmsClient: new KmsClient({httpsAgent}),
+    }),
+    hmac: new Hmac({
+      id: credentialsEdvHmacZcap.invocationTarget.id,
+      type: credentialsEdvHmacZcap.invocationTarget.type,
+      capability: credentialsEdvHmacZcap,
+      invocationSigner,
+      kmsClient: new KmsClient({httpsAgent}),
+    })
+  });
+
+  // FIXME: need edvClient.ensureIndex, what are indexes on this collection?
+  // these indexes came from VerifiableCredentialStore.js in
+  // bedrock-web-vc-store
+  edvClient.ensureIndex({attribute: ['meta.issuer', 'content.type']});
+  edvClient.ensureIndex({attribute: 'content.id', unique: true});
```

Comment:
I think these are ok for now. Let's role with it and adjust later. Ideally, we'll start expressing these in the profile's user doc that has some edv configuration info and we can just use that list, so it's all in one place.